SU2_CFD/src/solvers/CIncNSSolver.cpp

```diff
@@ -366,3 +378,304 @@ void CIncNSSolver::BC_ConjugateHeat_Interface(CGeometry *geometry, CSolver **sol
     nodes->SetEnergy_ResTruncError_Zero(iPoint);
   }
 }
+
+
+
+void CIncNSSolver::SetTauWall_WF(CGeometry *geometry, CSolver **solver_container, CConfig *config) {
+
+  /*---  The wall function implemented herein is based on Nichols and Nelson AIAAJ v32 n6 2004.
+   At this moment, the wall function is only available for adiabatic flows.
+   ---*/
+  su2double RefDensity,RefVel2; 
+  if ((config->GetRef_Inc_NonDim() == DIMENSIONAL) || (config->GetRef_Inc_NonDim() == INITIAL_VALUES)) {
+    RefDensity = Density_Inf;
+    RefVel2 = 0.0;
+    for (auto iDim = 0u; iDim < nDim; iDim++) 
+      RefVel2 += Velocity_Inf[iDim] * Velocity_Inf[iDim];
+  } else if (config->GetRef_Inc_NonDim() == REFERENCE_VALUES) {
+    RefDensity = config->GetInc_Density_Ref();
+    RefVel2 = config->GetInc_Velocity_Ref() * config->GetInc_Velocity_Ref();
+  }
+
+  bool debug = true;
+  unsigned long counter;
+
+  su2double UnitNormal[3];
+  su2double **grad_primvar, tau[3][3];
+  su2double TauElem[MAXNDIM] = {0.0};
+
+  su2double VelTangMod, WallDistMod;
+  su2double T_Normal, P_Normal;
+  su2double Density_Wall, T_Wall, P_Wall, Lam_Visc_Wall, Tau_Wall = 0.0;
+  su2double diff, Delta, grad_diff;
+  su2double U_Tau, U_Plus, Gam, Beta, Phi, Q, Y_Plus_White, Y_Plus;
+  su2double TauNormal, WallShearStress;
+  su2double Gas_Constant = config->GetGas_ConstantND();
+  su2double Cp = (Gamma / Gamma_Minus_One) * Gas_Constant;
+  su2double Eddy_Visc;
+  su2double Vector[MAXNDIM] = {0.0};
+  unsigned short max_iter = 100;
+  su2double tol = 1e-12;
+
+  /*--- Get the freestream velocity magnitude for non-dim. purposes ---*/
+
+  su2double *VelInf = config->GetVelocity_FreeStreamND();
+  su2double VelInfMod = 0.0;
+  for (auto iDim = 0u; iDim < nDim; iDim++)
+    VelInfMod += VelInf[iDim];
+  VelInfMod = sqrt(VelInfMod);
```

Comment:
```suggestion
  /*--- Get the freestream velocity magnitude for non-dim. purposes ---*/

const su2double VelInfMod = GeometryToolbox::Norm(nDim, config->GetVelocity_FreeStreamND());
```
Not sure if `VelInfMod += VelInf[iDim];` is correct here or if it should be the squared velinf

Reply:
it was already there and is not used, I will remove it